dfsclient/src/test/java/org/hpccsystems/dfs/client/DFSReadWriteTest.java

```diff
@@ -854,6 +862,33 @@ private void writeFile(List<HPCCRecord> records, String fileName, FieldDef recor
         }
     }
 
+    @Test
+    public void readIndexTest() throws Exception
+    {
+        String indexName = "test::index::child_dataset::key";
+
+        String eclFile = "child-index.ecl";
```

Comment:
let's externalize this functionality, maybe a function in BaseRemoteTest which loads and execute the ecl based on file name. we could execute the appropriate ecl once before the test run, and the tests should be skipped if the required datasets are not available.